### PR TITLE
feat(norm): add default component versioning

### PR DESCRIPTION
**Pull request by temich**
Component's directory hash is the default value for `version`.

### CODE REVIEW

File: runtime/norm/src/.component/.expand/version.js [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
+
+    stream.pipe(hash, { end: false })
+
+    return await once(stream, 'end')
   }
 
+  if (stat.isDirectory()) {
+    const entries = await fs.readdir(path)
+
+    for (const entry of entries) {
```
Comment (tinovyatkin) [inline runtime/norm/src/.component/.expand/version.js]: NIT: [fs.opendir](https://nodejs.org/api/fs.html#fspromisesopendirpath-options) will be more memory efficient here as we are doing stream way

File: runtime/norm/src/.component/.expand/version.js [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
+      await hashd(join(path, entry), hash)
+    }
+
```
Comment (tinovyatkin) [inline runtime/norm/src/.component/.expand/version.js:41]: This implementation will completely ignore symlinks (`isSymbolicLink()` case) and that is how some packages managers like `pnpm` or `yarn` are installing dependencies nowadays. Is that by design?

File: operations/src/deployment/images/image.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+  posix
+} = require('node:path')
+const {
+  readFile: read,
+  writeFile: write
+} = require('node:fs/promises')
+const { createHash } = require('node:crypto')
+
+const { overwrite } = require('@toa.io/generic')
 const fs = require('fs-extra')
```
Comment (tinovyatkin) [inline operations/src/deployment/images/image.js]: NIT: unrelated by this PR but as you are here - we can can get rid of `fs-extra` from this module as it only used for `ensureDir` which is available natively as `fs.mkdir(name, { recursive: true })`

File: runtime/norm/src/.component/.expand/version.js (hunk 2 above)
Comment (temich) [reply runtime/norm/src/.component/.expand/version.js:41]: This reveals deeper problem.

While components usually being developed inside a monorepo, thus their dependencies are pinned in the root level `package-lock`, when they are [installed inside docker image](https://github.com/toa-io/toa/blob/f28d629a9477646e267a8af8479cc1bb10d62c80/operations/src/deployment/images/composition.Dockerfile#L16), this `package-lock` doesn't exists.

At this point I think we may assume that if there are no changes in `package.json`, then previously built image may be used for deployment.

Unless you have some other suggestions.

Comment (tinovyatkin) [reply runtime/norm/src/.component/.expand/version.js:41]: I do agree that it probably OK for our case. People with complex deps can either pin exact dependency version or manage component versioning with lerna.

**Review by tinovyatkin — APPROVED** [review]

**Pull request merged**